src/Lean/Elab/InfoTree/Types.lean

```diff
@@ -224,6 +241,8 @@ inductive Info where
   | ofFieldRedeclInfo (i : FieldRedeclInfo)
   | ofDelabTermInfo (i : DelabTermInfo)
   | ofChoiceInfo (i : ChoiceInfo)
+  | ofDocInfo (i : DocInfo)
+  | ofDocElabInfo (i : DocElabInfo)
```

Comment:
I don't think I quite understand this distinction yet. Why are both of these needed, and what purpose does each serve?

Would it be possible to combine these?

Reply:
They indicate different things.

`DocInfo` indicates that ordinary syntax was elaborated as documentation - things like paragraphs and ordered lists. This is the non-extensible part of the documentation elaboration setup. It's used to recognized docstrings/parts of them that have actually been elaborated.

`DocElabInfo` indicates that a particular user-extensible documentation elaborator was used, such as `lean` to indicate Lean code. It's used similarly to how `TermInfo` is used to create hovers, but using `TermInfo` here gives misleading information.

I'll update with docstrings.